staging/src/k8s.io/component-base/cli/flag/string_slice_flag.go

```diff
@@ -45,9 +46,11 @@ func (s *StringSlice) String() string {
 }
 
 func (s *StringSlice) Set(val string) error {
-	if s.value == nil || !s.changed {
-		v := make([]string, 0)
-		s.value = &v
+	if s.value == nil {
+		return fmt.Errorf("no target (nil pointer to []string)")
+	}
+	if *s.value == nil || !s.changed {
+		*s.value = make([]string, 0)
```

Comment:
append already does this if the first arg is nil, so I don't think you need the first clause in the `if` at least

Reply:
As far as I can think of, this is indeed redundant. But I paid attention to similar codes, they all wrote this way, so I kept the first clause in the `if`.

https://github.com/kubernetes/kubernetes/blob/48b7492f92997ac8cbb8b6c559a0c68cba3021ab/staging/src/k8s.io/component-base/cli/flag/map_string_string.go#L69